app/components/login_button_component.rb

```diff
@@ -0,0 +1,29 @@
+class LoginButtonComponent < ButtonComponent
+  VALID_COLORS = ['primary', 'primary-darker', 'primary-lighter'].freeze
+
+  attr_reader :color, :tag_options
+
+  def initialize(color: 'primary', **tag_options)
+    if !VALID_COLORS.include?(color)
+      raise ArgumentError, "`color` #{color}} is invalid, expected one of #{VALID_COLORS}"
+    end
+
+    super(
+      **tag_options
+    )
+
+    @color = color
+  end
+
+  def logo_class
+    return 'login-button__logo login-button__logo-white' if color == 'primary-darker'
+    'login-button__logo'
+  end
```

Comment:
With the revisions from my [earlier comment](https://github.com/18F/identity-idp/pull/9840#discussion_r1446548621), we don't need this method anymore.

```suggestion
```